src/rsa/signing.rs

```diff
@@ -176,10 +213,18 @@ impl RSAKeyPair {
                     return Err(error::Unspecified);
                 }
 
+                // [NIST SP-800-56B rev. 1]] 6.4.1.4.3 - Step 6
+                //
+                // TODO 6.a: d < LCM(p – 1, q – 1)
+                // TODO 6.b: 1 == (d · e) mod LCM(p – 1, q – 1)
+                //
+                // [NIST SP-800-56B rev. 1]] 6.4.1.4.3 - Step 6.a, partial
                 if !(d.bit_length() > half_n_bits) {
```

Comment:
I know I am the one that initially said the check should be `d.bit_length() > half_n_bits`, but is this actually mathematically consistent with what NIST says? Should it be `d.bit_length() >= half_n_bits` instead? It would be great to write the proof here regarding why the `bit_length()` check makes sense.

Note: It may be helpful to use related facts that that we know, such as `e.bit_length() <= 33` and/or that `d` is odd.